### DISCUSSION TITLE
Non-Blocking Invocation in SignalR with MessagePack encoding

### DISCUSSION THREAD

**Discussion by MerzMax** [dotnet/aspnetcore#23379] [General] [answered]
I think there might be a mistake in the docs for the [SignalR Hub Protocol](https://github.com/dotnet/aspnetcore/blob/master/src/SignalR/docs/specs/HubProtocol.md). 

If an invocation isn't blocking the docs mentions that the Caller doesn't have to send an `InvocationId` with his request. 

In the encoding for the MessagePack Invocation message is a parameter called `NonBlocking`. Its function is never mentioned and the examples don't use it as well. 

The question I am asking myself is, if this parameter is used in any scenario and if, which value it might have.

**Comment by BrennanConroy** [comment, marked as answer]
Was just a previous piece of the spec from before the protocol was finalized.
Fixed in https://github.com/dotnet/aspnetcore/pull/23389

**Reply by MerzMax** [reply]
All right. Thanks for fixing the docs.
